src/web/routes.rs

```diff
@@ -270,6 +270,7 @@ impl Routes {
                 pattern.to_string(),
                 Box::new(RequestRecorder::new(
                     SimpleRedirect::new(|url| {
+                        #[allow(clippy::unnecessary_to_owned)]
                         url.set_path(&url.path().trim_end_matches('/').to_string())
```

Comment:
Hmm, why doesn't removing `to_string` work?

Reply:
```
error[E0502]: cannot borrow `*url` as mutable because it is also borrowed as immutable
   --> src/web/routes.rs:273:25
    |
273 |                         url.set_path(&url.path().trim_end_matches('/'))
    |                         ^^^^--------^^----------^^^^^^^^^^^^^^^^^^^^^^^
    |                         |   |         |
    |                         |   |         immutable borrow occurs here
    |                         |   immutable borrow later used by call
    |                         mutable borrow occurs here
```

if I'm not missing anything, we cannot get around this, right?

Reply:
ah, makes sense - maybe report this upstream to clippy as a bug?

Reply:
will do that, when I can use clippy locally again and prepare a proper issue. 

( I'm a victim of https://github.com/rust-lang/rust-clippy/issues/8470 )